provider/internal/java/dependency.go

```diff
@@ -195,7 +195,6 @@ func (p *javaServiceClient) GetDependenciesDAG(ctx context.Context) (map[uri.URI
 	args := []string{
 		"dependency:tree",
 		"-Djava.net.useSystemProxies=true",
-		fmt.Sprintf("-DoutputFile=%s", f.Name()),
```

Comment:
If you are not putting it out to a file, then I think we should be able to remove the creation of the temp file and the defer from above.